src/cleaner/scripts/clean_docker.py

```diff
@@ -54,44 +55,78 @@ def run(self):
-            self.kill_largest_container()
+            self.kill_largest_container(sized, used, usep)
 
 
     # Clean logic v1: kill largest container
     white_list = ["k8s_POD", "k8s_kube", "k8s_pylon", "k8s_zookeeper", "k8s_rest-server", "k8s_yarn", "k8s_hadoop", "k8s_job-exporter", "k8s_watchdog", "k8s_grafana", "k8s_node-exporter", "k8s_webportal", "k8s_prometheus", "k8s_nvidia-drivers", "k8s_etcd-container", "k8s_apiserver-container", "k8s_docker-cleaner", "kubelet", "dev-box"]
-    def kill_largest_container(self):
+    def kill_largest_container(self, sized, used, usep):
         containers = []
         # Only try to stop PAI jobs and user created containers
-        containers_source = subprocess.Popen(["docker", "ps", "-a", "--format", r'{{.ID}}\t{{.Image}}\t{{.Size}}\t{{.Names}}'], stdout=subprocess.PIPE)
+        containers_source = subprocess.Popen(["docker", "ps", "-a", "--format", r'{{.ID}}\t{{.Image}}\t{{.Size}}\t{{.Names}}\t'], stdout=subprocess.PIPE)
         for line in containers_source.stdout:
             splitline = line.split("\t")
             for prefix in self.white_list:
                 if (splitline[3].startswith(prefix)):
                     break
             else:
-                size = common.calculate_size(splitline[2].split()[0])
-                containers.append([size, splitline[0], splitline[1]])
+                size_str = splitline[2].split()[0]
+                size = common.calculate_size(size_str)
+                containers.append([size, splitline[0], splitline[1], splitline[3], size_str])
 
         containers.sort(key=lambda x:x[0], reverse=True)
 
         if containers.count > 0 and containers[0][0] > 1024**3:
-            self.logger.warning("Kill container {0} due to disk pressure. Container size: {1}".format(containers[0][1], containers[0][0]))
+            self.logger.warning("Kill container {0} due to disk pressure. Container size: {1}".format(containers[0][3], containers[0][4]))
+            
+            # Write error log
+            container_name = re.search(r"container(_\w+)?_\d+_\d+_\d+_\d+$", containers[0][3]).group()
+            application_name = "application{0}".format(re.search(r"^_\d+_\d+", re.search(r"_\d+_\d+_\d+_\d+$", container_name).group()).group())            
+            full_path = "/logs/{0}/{1}".format(application_name, container_name)
+
+            if not os.path.isdir(full_path):
+                self.logger.error("Cannot find job log dir, creating path. Log may not be collected.")
+                try:
+                    os.makedirs(full_path)
+                except OSError as exc:
+                    self.logger.error("Failed to create path {0}.".format(full_path))
+
+            if os.path.isdir(full_path):
+                error_filename = "{0}/diskCleaner.pai.error".format(full_path)
+                timestamp = int(time.time())
+                try:
+                    fp = open(error_filename, "w")
```

Comment:
please use `with` instead, otherwise the `else` block may still raise exception, such as 
```python
try:
    with open("file", "w") as f:
        f.writelines("")
except IOError:
    logger.error("")
```